Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -76,19 +81,47 @@
             "visibleRule": "DeployToSlotFlag = true"
         },
         {
+            "name": "VirtualApplication",
+            "type": "string",
+            "label": "Virtual Application",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Specify the name of the Virtual Application that has been configured in the Azure portal. The option is not required for deployments to the website root."
+        },
+        {
             "name": "Package",
             "type": "filePath",
-            "label": "Package",
+            "label": "Package or Folder",
             "defaultValue": "",
             "required": true,
             "helpMarkDown": "Location of the Web App zip package on the automation agent or on a UNC path accessible to the automation agent like, \\\\\\\\BudgetIT\\Web\\Deploy\\Fabrikam.zip. Predefined system variables and wild cards like, $(System.DefaultWorkingDirectory)\\\\***.zip` can be also used here."
         },
         {
+            "name": "WebAppUri",
+            "type": "string",
+            "label": "Web App Hosted Url",
+            "required": false,
+            "defaultValue": "",
+            "groupName": "output",
+            "helpMarkDown": "Provide a name for the variable for the webapp hosted url. The variable can be used as $(variableName) to refer to the hosted url of webapp in subsequent tasks like in the PowerShell on Target Machines task."
+        },
+        {
+            "name": "UseWebDeploy",
+            "type": "boolean",
+            "label": "Publish using Web Deploy",
+            "required": false,
+            "defaultValue": "true",
+            "groupName": "AdditionalDeploymentOptions",
+            "helpMarkDown": "The Web Deploy options will only work when the automation agent is on Windows and will be ignored on Linux. The options are also ignored for file formats that are not zipped or in a  folders, like Java WAR files. FTP works with automation agent on both Windows and Linux and for zip files, folders, and Java WAR files."
```

Comment:
"Publish using web deploy options are supported only when using Windows agent. On other platforms, the task relies on Kudu REST APIs (https://github.com/projectkudu/kudu/wiki/REST-API) to deploy the Web App, and following options are not supported"